lib/client.js

```diff
@@ -792,62 +793,46 @@ class Client extends events.EventEmitter {
-            !execOptions.isPrepared() &&
+            // !execOptions.isPrepared() &&
             params &&
-            !Array.isArray(params) &&
-            !types.protocolVersion.supportsNamedParameters(version)
+            !Array.isArray(params)
+            // && !types.protocolVersion.supportsNamedParameters(version)
         ) {
-            // Only Cassandra 2.1 and above supports named parameters
-            throw new errors.ArgumentError(
-                "Named parameters for simple statements are not supported, use prepare flag",
-            );
+            throw new Error(`TODO: Implement any support for named parameters`);
+            // // Only Cassandra 2.1 and above supports named parameters
+            // throw new errors.ArgumentError(
+            //   "Named parameters for simple statements are not supported, use prepare flag",
+            // );
         }
 
-        let request;
-
         if (!this.connected) {
-            // Micro optimization to avoid an async execution for a simple check
+            // TODO: Check this logic and decide if it's needed. Probably do it while implementing (better) connection
+            // // Micro optimization to avoid an async execution for a simple check
             await this.#connect();
         }
 
-        if (!execOptions.isPrepared()) {
-            return new Promise(async (resolve, reject) => {
-                let rusty = await this.rustClient.queryUnpagedNoValues(query);
-                let x = new ResultSet(rusty);
-                resolve(x);
-            });
-            request = await this.#createQueryRequest(
-                query,
-                execOptions,
-                params,
-            );
-        } else {
-            throw new Error(`TODO: Not implemented`);
-            /* const lbp = execOptions.getLoadBalancingPolicy();
-    
-            // Use keyspace from query options if protocol supports per-query keyspace, otherwise use connection keyspace.
-            const queryKeyspace =
-                (types.protocolVersion.supportsKeyspaceInRequest(version) &&
-                    execOptions.getKeyspace()) ||
-                this.keyspace;
-    
-        const { queryId, meta } = await PrepareHandler.getPrepared(this, lbp, query, queryKeyspace);
-        request = await this.#createExecuteRequest(query, queryId, execOptions, params, meta);
-        */
-        }
-
-        return await RequestHandler.send(request, execOptions, this);
+        return new Promise(async (resolve, reject) => {
+            try {
+                let statement = await this.rustClient.prepareStatement(query);
+                let parsedParams = parseParams(statement, params);
+                let rusty = await this.rustClient.querySimple(
+                    statement,
+                    parsedParams,
+                );
+                resolve(new ResultSet(rusty));
+            } catch (e) {
+                reject(e);
+            }
+        });
     }
```

Comment:
I don't see the previous code use `reject` at all, and you do. Please explain it in a comment.

Reply:
It's simply me not knowing JS. Nvm.